testnet/src/event_dispatcher.rs

```diff
@@ -86,11 +85,40 @@ impl EventObserver {
         }).to_string();
 
         // Send payload
-        let res = stream.write_bufs(&vec![payload.as_bytes().into()]);
-        if let Err(err) = res {
-            error!("Event dispatcher failed sending buffer: {:?}", err);
-            panic!();
-        }
+        let mut retry_count = 0;
+        let mut backoff: f64 = 1.0;
+        let mut rng = thread_rng();
+        loop {
+            let response = Client::new()
+                .post(&self.endpoint)
+                .json(&payload)
+                .send();
+
+            match response {
+                Ok(response) => {
+                    if response.status() == StatusCode::OK {
+                        break;
+                    }
+                    error!("Event dispatcher: POST {:?} failed with error {:?}", self.endpoint, response);
+                },
+                Err(e) => {
+                    error!("Event dispatcher: POST {:?} failed with error {:?}", self.endpoint, e);
+                }
+            };
+
+            if retry_count >= self.retry_count {
+                error!("Event dispatcher: exponential backoff failed");
+                panic!();                   
+            }
+
+            retry_count += 1;
+            backoff = (2.0 * backoff + (backoff * rng.gen_range(0.0, 1.0))).min(60.0);
+            let sleep_sec = backoff as u64;
+            let sleep_nsec = (((backoff - (sleep_sec as f64)) as u64) * 1_000_000) as u32;
+            let duration = Duration::new(sleep_sec, sleep_nsec);
```

Comment:
That's a lot of type coercions -- it's probably better to just use `Duration::from_millis`, like:

```rust
let duration = Duration::from_millis((backoff * 1_000) as u64)
```

Or, if you _really_ want the precision, use `from_nanos`

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1535/commits/851bec83c3cbae8515867f260e109da6f4c744d7